### PR TITLE
Add Visualization for marti_nav_msgs::TrackedObjectArray and marti_nav_msgs::ObstacleArray

### PR DESCRIPTION
**Pull request by matt-attack**
Note this is branched off of another change, so that should be merged first.

### CODE REVIEW

File: mapviz_plugins/include/mapviz_plugins/object_plugin.h [hunk 1]
```diff
@@ -0,0 +1,138 @@
+// *****************************************************************************
+//
+// Copyright (c) 2014, Southwest Research Institute® (SwRI®)
```
Comment (pjreed) [inline mapviz_plugins/include/mapviz_plugins/object_plugin.h:3]:
```suggestion
// Copyright (c) 2020, Southwest Research Institute® (SwRI®)
```

File: mapviz_plugins/src/object_plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,467 @@
+// *****************************************************************************
+//
+// Copyright (c) 2014, Southwest Research Institute® (SwRI®)
```
Comment (pjreed) [inline mapviz_plugins/src/object_plugin.cpp:3]:
```suggestion
// Copyright (c) 2020, Southwest Research Institute® (SwRI®)
```

File: mapviz_plugins/src/object_plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+            SLOT(SetColor(const QColor&)));
+
+    startTimer(1000);
+  }
+
+  ObjectPlugin::~ObjectPlugin()
+  {
+  }
+
+  void ObjectPlugin::showIdsToggled(bool value)
```
Comment (pjreed) [inline mapviz_plugins/src/object_plugin.cpp:78]: Does this function need to exist?

File: mapviz_plugins/src/object_plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+    canvas_ = canvas;
+    SetColor(ui_.color->color());
+
+    return true;
+  }
+
+  void ObjectPlugin::Draw(double x, double y, double scale)
+  {
+    auto objIter = objects_.begin();
+    while (objIter != objects_.end())
```
Comment (pjreed) [inline mapviz_plugins/src/object_plugin.cpp:308]: Using a while loop and manually incrementing the iterator seems like a weird way to do this, any reason to not use a for-each loop?

File: mapviz_plugins/src/object_plugin.cpp [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+
+      if (!obj.transformed) {
+        objIter++;
+        continue;
+      }
+
+      glColor4f(color_.redF(), color_.greenF(), color_.blueF(), 1.0);
+
+      //glColor4f(marker.color.r, marker.color.g, marker.color.b, marker.color.a);
+
```
Comment (pjreed) [inline mapviz_plugins/src/object_plugin.cpp:320]:
```suggestion
```

File: mapviz_plugins/src/object_plugin.cpp [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+      glLineWidth(3.0);//marker.scale_x);
+      glBegin(GL_LINE_STRIP);
+
+      for (const auto &point : obj.polygon) {
+        //glColor4f(point.color.r, point.color.g, point.color.b, point.color.a);
+
```
Comment (pjreed) [inline mapviz_plugins/src/object_plugin.cpp:326]:
```suggestion
```

File: mapviz_plugins/src/object_plugin.cpp [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+    // translated appropriately.  So, we save off the current world transform
+    // and reset it; when we actually draw the text, we'll manually translate
+    // it to the right place.
+    QTransform tf = painter->worldTransform();
+    QFont font("Helvetica", 10);
+    painter->setFont(font);
+    painter->save();
+    painter->resetTransform();
+
+    for (auto objIter = objects_.begin(); objIter != objects_.end(); ++objIter)
```
Comment (pjreed) [inline mapviz_plugins/src/object_plugin.cpp:361]: Seems like the loops here and in Transform below could be for-each loops as well.

File: mapviz_plugins/src/object_plugin.cpp (hunk 3 above)
Comment (matt-attack) [reply mapviz_plugins/src/object_plugin.cpp]: Oh, no. Good catch.

**Comment by matt-attack** [comment]
@pjreed should be resolved now

**Pull request merged**